resolver-dns/src/main/java/io/netty/resolver/dns/TcpDnsQueryContext.java

```diff
@@ -29,14 +30,14 @@
 
 final class TcpDnsQueryContext extends DnsQueryContext {
 
-    TcpDnsQueryContext(Channel channel, Future<? extends Channel> channelReadyFuture,
+    TcpDnsQueryContext(ChannelFuture channelFuture,
```

Comment:
TCP doesn't suffer from observer conntrack race, and for TCP the cost of `DnsResolveNewChannelProvider` will be much higher. Can we make the provider configurable for UDP and TCP separately?

Reply:
The provider is only used for Udp

Reply:
I see, TCP always creates a new connection and closes it after query completes